lib/luatest_server.py

```diff
@@ -0,0 +1,110 @@
+import glob
+import os
+import re
+import sys
+from subprocess import STDOUT
+from gevent.subprocess import Popen, PIPE
+
+from lib.preprocessor import TestState
+from lib.sampler import sampler
+from lib.server import Server
+from lib.tarantool_server import Test
+from lib.tarantool_server import TarantoolServer
+
+
+class LuatestTest(Test):
+    """ Handle *_test.lua.
+
+    Provide method for executing luatest <name>_test.lua test.
+    """
+
+    def __init__(self, *args, **kwargs):
+        Test.__init__(self, *args, **kwargs)
+        self.valgrind = kwargs.get('valgrind', False)
+
+    def execute(self, server):
+        """Execute test by luatest command
+
+        Execute 'luatest -v <name>_test.lua -o tap --shuffle none'
+        Provide a verbose output in the tap format.
+        Use shuffle option in none mode for avoiding mixing tests.
+        Use capture mode.
+        """
+        server.current_test = self
+        ts = TestState(self.suite_ini, None, TarantoolServer,
+                       self.run_params,
+                       default_server_no_connect=server)
+        self.inspector.set_parser(ts)
+        command = [os.environ['LUATEST_BIN'], '-v', '-c', self.name,
+                   '-o', 'tap',
+                   '--shuffle', 'none']
+        proc = Popen(command, cwd=server.vardir, stdout=PIPE, stderr=STDOUT)
```

Comment:
I'm not very fun about mixing stdout in TAP13 format and stderr in arbitrary format. It is possible that the mix will anyway be parseable as TAP13 (or at least be accepted by given TAP13 parser). Say, if stderr is formatted in YAML, it may be interpreted as a [TAP13 YAML block](https://testanything.org/tap-version-13-specification.html#yaml-blocks). Depending of certain parser there may be other kinds of output that will be accepted. For example, our TAP13 parser [is not very strict](https://github.com/tarantool/test-run/blob/ab453e33960a02dc77e59cb71257486005d7891f/lib/pytap13.py#L167-L168) (`self.strict` is always `False`). AFAIR, I had intention to make the parser strictier, but our tests were not ready for that, so I hid this behaviour under the flag. It would be good to re-check, whether we can enable it now (out of this issue). There were fixes around incorrect test plans.

Back to mixing of stderr into stdout. In the past, AppTest had this behaviour, but I [changed](https://github.com/tarantool/test-run/commit/dac6b2b4640a7dd3d130ddde13bb4b25a730a4ad) it. I think it would be good to don't introduce it again for LuatestTest.

Reply:
Have to change `-c` option in luatest
https://github.com/tarantool/luatest/issues/147

Reply:
But the code in test-run that mixing stdout and stderr still there. It should not.

Reply:
I'll leave it as is for now.